Simplify GrpcOffersService#createOffer (refactor)

ghubstan opened the pull request:
Push the placeOffer call down into `CoreOffersService#createOffer`, and use a countdown latch to wait for async placeOffer to complete.

ghubstan committed:
Simplify GrpcOffersService#createOffer (refactor)

Push the placeOffer call down into CoreOffersService#createOffer,
and use a countdown latch to wait for async placeOffer to
complete.

ghubstan committed:
Fix method name

ghubstan committed:
Use consumer::accept instead of a countdown latch

This is more readable and stylistically consistent.

chimp1984 commented on core/src/main/java/bisq/core/api/CoreOffersService.java:
```diff
@@ -87,15 +88,16 @@ public CoreOffersService(CreateOfferService createOfferService,
-                      long minAmountAsLong,
-                      double buyerSecurityDeposit,
-                      String paymentAccountId) {
+    Offer createAndPlaceOffer(String currencyCode,
```
I think it would be more clear to not return the offer, as it is passed via the `Consumer<Offer> resultHandler`.

ghubstan replied:
Clever.   Fix is in commit 30e6dae.

ghubstan committed:
Change return type to void

The prior Offer return type is redundant and confusing given the
new offer is passed up via the Consumer<Offer> resultHandler.

chimp1984 reviewed (approved):
utACK

ripcurlx reviewed (approved):
utACK
